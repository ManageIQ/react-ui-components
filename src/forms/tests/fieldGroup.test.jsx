import React from 'react';
import { shallow } from 'enzyme';
import { Field, Form } from 'react-final-form';
import { required } from 'redux-form-validators';
import PropTypes from 'prop-types';
import toJson from 'enzyme-to-json';

import { FieldGroup, FinalFormRadio } from '../';

describe('Final form FieldGroup component', () => {
  const Group = ({ validateOnMount }) => (
    <Form
      onSubmit={() => {}}
      render={() => (
        <FieldGroup name="some group" label="Field group" validateOnMount={validateOnMount}>
          <Field
            name="radioGroup"
            id="selectOne1"
            type="radio"
            value="First radio"
            validate={required({ msg: 'Please select one option' })}
            render={({ input, meta, ...rest }) => <FinalFormRadio input={input} label="Radio button 1" meta={meta} {...rest} />}
          />
          <Field
            name="radioGroup"
            id="selectOne2"
            type="radio"
            value="Second radio"
            validate={required({ msg: 'Please select one option' })}
            render={({ input, meta, ...rest }) => <FinalFormRadio input={input} label="Radio button 2" meta={meta} {...rest} />}
          />
        </FieldGroup>
      )}
    />
  );

  Group.propTypes = {
    validateOnMount: PropTypes.bool,
  };

  it('Should render correctly', () => {
    const tree = shallow(<Group />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('Should render correctly with error state', () => {
    const tree = shallow(<Group validateOnMount />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
