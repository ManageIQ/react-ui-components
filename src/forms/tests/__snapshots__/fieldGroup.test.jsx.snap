// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Final form FieldGroup component Should render correctly 1`] = `
<ReactFinalForm
  onSubmit={[Function]}
  render={[Function]}
/>
`;

exports[`Final form FieldGroup component Should render correctly with error state 1`] = `
<ReactFinalForm
  onSubmit={[Function]}
  render={[Function]}
/>
`;
